rcl/src/rcl/logging.c

```diff
@@ -107,6 +112,12 @@ rcl_logging_configure(const rcl_arguments_t * global_args, const rcl_allocator_t
 rcl_ret_t rcl_logging_fini()
 {
   rcl_ret_t status = RCL_RET_OK;
+  uint64_t current_count = rcutils_atomic_fetch_add_uint64_t(&__rcl_logging_init_count, -1);
```

Comment:
using atomic is not exactly thread safe, but expecting good enough for this case. besides we do not really want to introduce mutex in here.